src/Engine.ts

```diff
@@ -302,110 +312,74 @@ export class Engine {
+          slapTrackers: this.slapTrackers
+        }
+        customBroadcasterConfig = {
+          resolver: new LookupResolver(resolverConfig)
         }
       }
+      const shipBroadcaster = new SHIPBroadcaster(relevantTopics, customBroadcasterConfig)
 
-      const broadcastPromises: Array<Promise<Response>> = []
-
-      // Make sure we gossip to the shipTrackers we know about.
-      if (this.shipTrackers !== undefined && this.shipTrackers.length !== 0 && relevantTopics.includes('tm_ship')) {
-        this.shipTrackers.forEach(tracker => {
-          if (domainToTopicsMap.get(tracker) !== undefined) {
-            domainToTopicsMap.get(tracker)?.add('tm_ship')
-          } else {
-            domainToTopicsMap.set(tracker, new Set(['tm_ship']))
-          }
-        })
+      try {
+        await shipBroadcaster.broadcast(tx)
+      } catch (error) {
+        console.error('Error during broadcasting:', error)
       }
 
-      // Make sure we gossip to the slapTrackers we know about.
-      if (this.slapTrackers !== undefined && this.slapTrackers.length !== 0 && relevantTopics.includes('tm_slap')) {
-        this.slapTrackers.forEach(tracker => {
-          if (domainToTopicsMap.get(tracker) !== undefined) {
-            domainToTopicsMap.get(tracker)?.add('tm_slap')
-          } else {
-            domainToTopicsMap.set(tracker, new Set<string>(['tm_slap']))
+      // Handle shipTrackers for 'tm_ship' topic
+      if (this.shipTrackers && this.shipTrackers.length > 0 && relevantTopics.includes('tm_ship')) {
+        const taggedBEEFForShip = {
+          beef: taggedBEEF.beef,
+          topics: ['tm_ship']
+        }
+        const facilitator = new HTTPSOverlayBroadcastFacilitator()
+
+        const sendPromises = this.shipTrackers.map(async tracker => {
+          try {
+            await facilitator.send(tracker, taggedBEEFForShip)
+          } catch (error) {
+            console.error(`Error sending to shipTracker ${tracker}:`, error)
           }
         })
+        await Promise.all(sendPromises)
       }
 
-      // Note: We are depending on window.fetch, this may not be ideal for the long term.
-      for (const [domain, topics] of domainToTopicsMap.entries()) {
-        if (domain !== this.hostingURL) {
-          const promise = fetch(`${String(domain)}/submit`, {
-            method: 'POST',
-            headers: {
-              'Content-Type': 'application/octet-stream',
-              'X-Topics': JSON.stringify(Array.from(topics))
-            },
-            body: new Uint8Array(taggedBEEF.beef)
-          })
-          broadcastPromises.push(promise)
+      // Handle slapTrackers for 'tm_slap' topic
+      if (this.slapTrackers && this.slapTrackers.length > 0 && relevantTopics.includes('tm_slap')) {
+        const taggedBEEFForSlap = {
+          beef: taggedBEEF.beef,
+          topics: ['tm_slap']
         }
-      }
+        const facilitator = new HTTPSOverlayBroadcastFacilitator()
```

Comment:
We might not want to restrict the Overlay Engine to HTTP. Maybe a new OverlayBroadcastFacilitator parameter could be introduced, allowing the injection of a preferred facilitator.

Reply:
Other than that, looks good!